File: core/src/commonMain/kotlin/com/encatch/core/DeviceInfo.kt
package com.encatch.core

internal const val SDK_VERSION = "0.1.0"
internal const val SDK_PLATFORM = "android"

/** Platform-sourced device facts, assembled into [ApiDeviceInfo] by [Encatch]. */
internal data class PlatformDeviceFacts(
    val osVersion: String,
    val deviceLocale: String,
    val timezone: String?,
    val appVersion: String?,
    val appPackageName: String?,
)

internal expect fun collectPlatformDeviceFacts(): PlatformDeviceFacts

File: ios-native/Sources/Encatch/Core/DeviceInfo.swift
import Foundation
#if canImport(UIKit)
import UIKit
#endif

let SDK_VERSION = "0.1.0"
let SDK_PLATFORM = "ios"

/// Platform-sourced device facts, assembled into `ApiDeviceInfo` by `Encatch`.
struct PlatformDeviceFacts {
    let osVersion: String
    let deviceLocale: String
    let timezone: String?
    let appVersion: String?
    let appPackageName: String?
}

func collectPlatformDeviceFacts() -> PlatformDeviceFacts {
    let bundle = Bundle.main
    let appVersion = bundle.infoDictionary?["CFBundleShortVersionString"] as? String
    let appPackageName = bundle.bundleIdentifier

    #if canImport(UIKit) && !os(watchOS)
    let osVersion = UIDevice.current.systemVersion
    #else
    let osVersion = ProcessInfo.processInfo.operatingSystemVersionString
    #endif

    return PlatformDeviceFacts(
        osVersion: osVersion,
        deviceLocale: Locale.current.identifier,
        timezone: TimeZone.current.identifier,
        appVersion: appVersion,
        appPackageName: appPackageName
    )
}
